knative-operator/pkg/controller/knativeserving/consoleclidownload/consoleclidownload.go

```diff
@@ -29,29 +29,28 @@ var log = common.Log.WithName("consoleclidownload")
 
 // Apply installs kn ConsoleCLIDownload and its required resources
 func Apply(instance *servingv1alpha1.KnativeServing, apiclient client.Client, scheme *runtime.Scheme) error {
-	if err := reconcileKnCCDResources(instance, apiclient, scheme); err != nil {
+	service := &servingv1.Service{}
+	if err := reconcileKnCCDResources(instance, apiclient, scheme, service); err != nil {
 		return err
 	}
-
-	if err := reconcileKnConsoleCLIDownload(apiclient, instance); err != nil {
+	if err := reconcileKnConsoleCLIDownload(apiclient, instance, service); err != nil {
 		return err
 	}
-
 	return nil
 }
 
 // reconcileKnCCDResources reconciles required resources viz Knative Service
 // which will serve kn cross platform binaries within cluster
-func reconcileKnCCDResources(instance *servingv1alpha1.KnativeServing, apiclient client.Client, scheme *runtime.Scheme) error {
+func reconcileKnCCDResources(instance *servingv1alpha1.KnativeServing, apiclient client.Client, scheme *runtime.Scheme, service *servingv1.Service) error {
 	if !instance.Status.IsReady() {
-		return fmt.Errorf("Knative instance %q/%q not ready yet", instance.Name, instance.GetNamespace())
+		// Don't return error, wait silently until Serving instance is ready
```

Comment:
This now won't have an effect as you'll pass an empty KnativeService down into the second function, which will then error because the service isn't ready yet.

I think we should pull this check to the top-level `Apply`

Reply:
I should follow my idea from https://github.com/openshift-knative/serverless-operator/pull/381#discussion_r497468806. :)

Reply:
Addressed in https://github.com/openshift-knative/serverless-operator/pull/381/commits/1725e378dd5bf945ee857089fffbd1d210be922c.